File: Algorithms/src/1266. Minimum Time Visiting All Points/Code/solution.py
class Solution:
    def minTimeToVisitAllPoints(self, points):
        total_time = 0

        for i in range(1, len(points)):
            dx = abs(points[i][0] - points[i - 1][0])
            dy = abs(points[i][1] - points[i - 1][1])

            total_time += max(dx, dy)

        return total_time

File: Algorithms/src/1266. Minimum Time Visiting All Points/Code/solution.js
/**
 * @param {number[][]} points
 * @return {number}
 */
var minTimeToVisitAllPoints = function (points) {
  let totalTime = 0;

  for (let i = 1; i < points.length; i++) {
    const dx = Math.abs(points[i][0] - points[i - 1][0]);
    const dy = Math.abs(points[i][1] - points[i - 1][1]);

    totalTime += Math.max(dx, dy);
  }

  return totalTime;
};
